src/components/file/file-metadata.spec.ts

```diff
@@ -0,0 +1,29 @@
+import { getFileIcon } from './file-metadata';
+
+describe('file metadata', () => {
+    describe('getFileIcon', () => {
+        it('returns the expected file type icon for a known extension', () => {
+            const iconName = getFileIcon({ id: '', filename: 'doc.docx' });
+            expect(iconName).toEqual('ms_word_copyrighted');
+        });
+        it('returns the default icon for unknown file type', () => {
+            const iconName = getFileIcon({
+                id: '',
+                filename: 'file.justsomethingrandom',
+            });
+            expect(iconName).toEqual('file');
+        });
+        it('returns the default icon for extensionless filename', () => {
+            const iconName = getFileIcon({
+                id: '',
+                filename: 'README',
+            });
+            expect(iconName).toEqual('file');
+        });
+        it('is case insensitive', () => {
+            const iconA = getFileIcon({ id: '', filename: 'doc.docx' });
+            const iconB = getFileIcon({ id: '', filename: 'DOC.DOCX' });
+            expect(iconA).toEqual(iconB);
```

Comment:
@adrianschmidt what do you think about this? something more that would be good to add while i'm at it?